src/selectors/form.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+/**
+ * Selects a boolean from the form state which is an indicator
+ * whether this form is in a valid and complete state.
+ * @return {bool}
+ */
+export const selectCanSaveForm = ({ form }) => {
+  const areAllValid = Object.values(form).every(({ isValid }) => isValid);
+  const allHaveValues = Object.values(form).every(({ hasValue }) => hasValue);
```

Comment:
:nit minor naming comment - to be strictly consistent `allAreValid` and `allHaveValues` 

I went down a small rabbit hole of wondering if it is ok to separate validity and 'has a value' - a required field is only valid if it has a value, and optional fields don't need a value, so the `hasValue` isn't required.